### PR TITLE
.netstandard 2.0

### LINKED ISSUE #977: Optimize slow running tests
Ideally we should aim for having all tests running in less than half second, but PRs can focus on each of those concern lists:

## CRITICAL - >= 5 seconds
- [ ] 30.540s - Toggl Foundation Tests MvvmCross ViewModels EditTimeEntryViewModelTests TheSelectTagsCommand (4 tests) 
- [ ] 8.745s - Toggl Foundation Tests Autocomplete AutocompleteProviderTests TheQueryMethod QueriesTheDatabaseForTimeEntries (10 tests) 
- [ ] 6.375s - Toggl Foundation Tests Sync States FetchAllSinceStateTests TheStartMethod (8 tests) 
- [ ] 6.326s - Toggl Foundation Tests Sync SyncManagerTests TheOrchestratorCompleteObservable (11 tests) 

## BAD - >= 2 seconds
- [ ] 5.704s - Toggl Foundation Tests Sync SyncManagerTests TheForceFullSyncMethod (10 tests) 
- [ ] 5.130s - Toggl Foundation Tests MvvmCross ViewModels SettingsViewModelTests TheLogoutCommand (15 tests) 
- [ ] 4.548s - Toggl Foundation Tests MvvmCross ViewModels TimeEntriesLogViewModelTests TheTimeEntriesProperty (3 tests) 
- [ ] 4.423s - Toggl Foundation Tests Autocomplete AutocompleteProviderTests TheQueryMethod QueriesTheDatabaseForProjects (6 tests) 
- [ ] 4.153s - Toggl Foundation Tests Sync SyncManagerTests ThePushSyncMethod (10 tests) 
- [ ] 3.476s - Toggl Foundation Tests MvvmCross ViewModels SuggestionsViewModelTests TheStartTimeEntryCommand (3 tests) 
- [ ] 3.134s - Toggl Foundation Tests MvvmCross ViewModels TimeEntriesLogViewModelTests TheContinueTimeEntryCommand (4 tests) 
- [ ] 3.005s - Toggl Foundation Tests Sync ConflictResolution TimeEntryRivalsResolverTests (8 tests) 
- [ ] 2.859s - Toggl Foundation Tests MvvmCross ViewModels SettingsViewModelTests TheFlags (6 tests) 
- [ ] 2.772s - Toggl Foundation Tests MvvmCross ViewModels EditTimeEntryViewModelTests TheInitializeMethod (4 tests) 
- [ ] 2.478s - Toggl Foundation Tests MvvmCross ViewModels SelectProjectViewModelTests TheSuggestionsProperty (6 tests) 
- [ ] 2.007s - Toggl Foundation Tests Sync TheDeadlocks TheStateMachine (4 tests) 

## LEAST CONCERN - >= 1 seconds
- [ ] 1.997s - Toggl Foundation Tests Login LoginManagerTests TheLoginMethod (20 tests) 
- [ ] 1.700s - Toggl Foundation Tests MvvmCross ViewModels EditTimeEntryViewModelTests TheTagsProperty (10 tests) 
- [ ] 1.630s - Toggl Foundation Tests MvvmCross ViewModels StartTimeEntryViewModelTests TheCreateProjectCommand (2 tests) 
- [ ] 1.555s - Toggl Foundation Tests Sync States Push UnsyncableUserStateTests (9 tests) 
- [ ] 1.462s - Toggl Foundation Tests Sync States Push UnsyncableTimeEntryStateTests (9 tests) 
- [ ] 1.371s - Toggl Foundation Tests Sync States Push UnsyncableProjectStateTests (9 tests) 
- [ ] 1.270s - Toggl Foundation Tests MvvmCross ViewModels StartTimeEntryViewModelTests TheChangeDurationCommandCommand (2 tests) 
- [ ] 1.262s - Toggl Foundation Tests MvvmCross ViewModels EditProjectViewModelTests TheCloseCommand (3 tests) 
- [ ] 1.248s - Toggl Foundation Tests Sync States Push UnsyncableClientStateTests (9 tests) 
- [ ] 1.202s - Toggl Foundation Tests Sync TheDeadlocks TheSyncManager (6 tests) 
- [ ] 1.178s - Toggl Foundation Tests Sync States Push UnsyncableTagStateTests (9 tests) 
- [ ] 1.155s - Toggl Foundation Tests MvvmCross ViewModels SelectWorkspaceViewModelTests TheSelectWorkspaceCommand (2 tests) 
- [ ] 1.108s - Toggl Ultrawave Tests ApiClients BaseApiTests TheGetAuthHeaderMethod (1 test) 
- [ ] 1.105s - Toggl Foundation Tests MvvmCross ViewModels EditTimeEntryViewModelTests TheConstructor (15 tests) 
- [ ] 1.063s - Toggl Foundation Tests MvvmCross ViewModels EditDurationViewModelTests ThePrepareMethod (4 tests) 
- [ ] 1.012s - Toggl Foundation Tests MvvmCross ViewModels StartTimeEntryViewModelTests TheSelectSuggestionCommand WhenSelectingAProjectSuggestion (14 tests) 

## PERFECTIONISM - >= 0.5 second
- [ ] 0.987s - Toggl Foundation Tests MvvmCross ViewModels EditDurationViewModelTests TheSaveCommand (3 tests) 
- [ ] 0.940s - Toggl Foundation Tests MvvmCross ViewModels StartTimeEntryViewModelTests TheToggleProjectSuggestionsCommand (27 tests) 
- [ ] 0.929s - Toggl Multivac Tests StringExtensionTests TheUnicodeSafeSubstringMethod (25 tests  
- [ ] 0.914s - Toggl Foundation Tests MvvmCross ViewModels StartTimeEntryViewModelTests TheDoneCommand StartsANewTimeEntry (13 tests) 
- [ ] 0.881s - Toggl PrimeRadiant Tests Realm RealmSingleObjectStorageTests (9 tests  
- [ ] 0.850s - Toggl Foundation Tests MvvmCross ViewModels StartTimeEntryViewModelTests TheToggleTagSuggestionsCommand (26 tests) 
- [ ] 0.833s - Toggl Foundation Tests Sync ConflictResolution PreferNewerTests (11 tests) 
- [ ] 0.825s - Toggl Foundation Tests DataSources WorkspacesDataSourceTests TheConstructor (1 test) 
- [ ] 0.804s - Toggl Foundation Tests MvvmCross ViewModels StartTimeEntryViewModelTests TheSelectSuggestionCommand WhenSelectingATimeEntrySuggestion (10 tests) 
- [ ] 0.774s - Toggl Foundation Tests MvvmCross ViewModels StartTimeEntryViewModelTests TheSelectSuggestionCommand WhenSelectingATaskSuggestion (14 tests) 
- [ ] 0.772s - Toggl Foundation Tests MvvmCross ViewModels SelectProjectViewModelTests TheCloseCommand (3 tests) 
- [ ] 0.727s - Toggl Foundation Tests MvvmCross ViewModels EditDurationViewModelTests TheDurationProperty (3 tests) 
- [ ] 0.719s - Toggl Multivac Tests StringExtensionTests TheContainsIgnoringCaseMethod (2 tests  
- [ ] 0.701s - Toggl Ultrawave Tests Models ProjectTests (4 tests)
- [ ] 0.679s - Toggl Foundation Tests DataSources TogglDataSourceTests TheHasUnsyncedDataMetod ProjectsTest (3 tests) 
- [ ] 0.674s - Toggl Foundation Tests Autocomplete AutocompleteProviderTests TheQueryMethod QueriesTheDatabaseForTags (5 tests) 
- [ ] 0.649s - Toggl Ultrawave Tests Models UserTests TheUserModel (3 tests) 
- [ ] 0.638s - Toggl Foundation Tests MvvmCross ViewModels EditTimeEntryViewModelTests TheEditDurationCommand (2 tests) 
- [ ] 0.628s - Toggl Ultrawave Tests Models ClientTests TheClientModel (3 tests) 
- [ ] 0.601s - Toggl Foundation Tests MvvmCross ViewModels SelectTagsViewModelTests TheTagsProperty (4 tests) 
- [ ] 0.591s - Toggl PrimeRadiant Tests Realm RealmRepositoryTests (9 tests  
- [ ] 0.566s - Toggl Ultrawave Tests Models TimeEntryTests TheTimeEntryModel (4 tests) 
- [ ] 0.563s - Toggl Foundation Tests Sync States UpdateTimeEntryStateTests (20 tests) 
- [ ] 0.544s - Toggl Foundation Tests MvvmCross ViewModels EditTimeEntryViewModelTests TheSelectProjectCommand (6 tests) 
- [ ] 0.544s - Toggl Ultrawave Tests Serialization JsonSerializerTests TheSerializeMethod (2 tests) 
- [ ] 0.534s - Toggl Foundation Tests DataSources TogglDataSourceTests TheHasUnsyncedDataMetod TagsTest (3 tests) 
- [ ] 0.532s - Toggl Foundation Tests DataSources TogglDataSourceTests TheHasUnsyncedDataMetod TimeEntriesTest (3 tests) 
- [ ] 0.529s - Toggl Ultrawave Tests ApiClients BaseApiTests TheCreateObservableMethod (2 tests) 
- [ ] 0.505s - Toggl Foundation Tests Sync States CreateTimeEntryStateTests (19 tests)

### PR DESCRIPTION
**Pull request by willsb**
Closes #977
Ref #1157

Things missing:

- [X] Make tests run in :cake:
- [x] Fix integration tests
- [x] Fix UI tests (ref #1149)
- [x] Update solution maps so we can use Release.AppStore

[Integration Tests](https://www.bitrise.io/build/3d749cd0b4d8c973)
[UI Tests](https://www.bitrise.io/build/2574636f13f415cc)

### CODE REVIEW

File: Toggl.Daneel/LinkerPleaseInclude.cs [hunk 1]
```diff
@@ -179,5 +180,18 @@ public void Include(MvxUISliderValueTargetBinding binding)
         {
             binding = new MvxUISliderValueTargetBinding(null, null);
         }
+
+        public void Include(ConsoleColor color)
```
Comment (willsb) [inline Toggl.Daneel/LinkerPleaseInclude.cs]: Ref: https://stackoverflow.com/a/47052169/3465182

File: Toggl.Multivac.Tests/EmailTests.cs [hunk 2]
```diff
@@ -92,7 +92,7 @@ public sealed class TheToFullNameMethod
             [InlineData("šimon@gmail.cz", "Šimon")]
             [InlineData("ägypter@gmail.de", "Ägypter")]
             [InlineData("леонтий@gmail.ru", "Леонтий")]
-            [InlineData("ıwithout.a.dot@gmail.com", "Iwithout A Dot")]
+            [InlineData("ıwithout.a.dot@gmail.com", "ıwithout A Dot")]
```
[the remark this answers is not in the document]
Comment (simonrozsival) [reply Toggl.Multivac.Tests/EmailTests.cs]: The test worked well and I was convinced that `ı` should be uppercased to `I`. Maybe there was some change in the way unicode is implemented in the runtime that is used after this change?

On the other hand, this is a quite ridiculous test, maybe it wasn't necessary before (it is not common to use special characters in the first part of an email address) and it should be rather 🔥 than fixing.

File: Toggl.Foundation.Tests/Sync/States/Push/PushTimeEntryStateTests.cs [hunk 3]
```diff
@@ -1,32 +0,0 @@
-﻿using Toggl.Foundation.Models;
-using Toggl.Foundation.Sync.States;
-using Toggl.PrimeRadiant.Models;
-
-namespace Toggl.Foundation.Tests.Sync.States
-{
-    public sealed class PushTimeEntryStateTests : BasePushEntityStateTests
```
[the remark this answers is not in the document]
Comment (willsb) [reply Toggl.Foundation.Tests/Sync/States/Push/PushTimeEntryStateTests.cs]: Unused file got 🔥

File: Toggl.Ultrawave.Tests/Models/WorkspaceTests.cs [hunk 4]
```diff
@@ -30,15 +30,15 @@ public void HasConstructorWhichCopiesValuesFromInterfaceToTheNewInstance()
             }
 
             [Theory, LogIfTooSlow]
-            [MemberData("SerializationCases")]
+            [MemberData(nameof(SerializationCases))]
```
[the remark this answers is not in the document]
Comment (willsb) [reply Toggl.Ultrawave.Tests/Models/WorkspaceTests.cs]: It does. The project won't compile without it because of the new xUnit analysers

File: Toggl.Multivac.Tests/EmailTests.cs [hunk 5]
```diff
@@ -92,8 +92,6 @@ public sealed class TheToFullNameMethod
             [InlineData("šimon@gmail.cz", "Šimon")]
             [InlineData("ägypter@gmail.de", "Ägypter")]
             [InlineData("леонтий@gmail.ru", "Леонтий")]
-            [InlineData("ıwithout.a.dot@gmail.com", "ıwithout A Dot")]
-            [InlineData("iwith.a.dot@gmail.com", "Iwith A Dot")]
```
Comment (amulware) [inline Toggl.Multivac.Tests/EmailTests.cs]: Why is this removed?

Comment (willsb) [reply Toggl.Multivac.Tests/EmailTests.cs]: I removed based on @simonrozsival's comments

Comment (amulware) [reply Toggl.Multivac.Tests/EmailTests.cs]: The first line sure, but the second?

Comment (willsb) [reply Toggl.Multivac.Tests/EmailTests.cs]: Oh, I see. That can be added back
